pkg/apis/triggers/v1beta1/event_listener_validation.go

```diff
@@ -149,6 +149,10 @@ func validateKubernetesObject(orig *KubernetesResource) (errs *apis.FieldError)
 		errs = errs.Also(validateEnv(orig.Template.Spec.Containers[0].Env).ViaField("spec.template.spec.containers[0].env"))
 	}
 
+	if orig.ServiceLoadBalancerClass != nil && orig.ServiceType != corev1.ServiceTypeLoadBalancer {
```

Comment:
Defining a `LoadBalancerClass` is just an optionnal parameter for LoadBalancers, so if we don't want people to customize `ServicePort` when they choose a LoadBalancer type, there is no interaction with the `LoadBalancerClass` field.

I can add the check if you think I should, but it's not linked to LoadBalancerClass.

Reply:
actually my question was if we set LoadBalancerClass then usually do we set servicePort also ?

Reply:
Most of the time it's not necessary but it may I guess.
The service port is a legitimate thing to customize for a loadbalancer service, so I think we shouldn't have a rule against it.

Reply:
Okay then that's fine :+1: